mono/metadata/class-accessors.c

```diff
@@ -389,3 +390,31 @@ mono_class_gtd_get_canonical_inst (MonoClass *klass)
 	g_assert (mono_class_is_gtd (klass));
 	return &((MonoClassGtd*)klass)->canonical_inst;
 }
+
+typedef struct {
+	MonoPropertyBagItem head;
+
+	int nbits;
+	gsize *bits;
+} WeakBitmapData;
+
+void
+mono_class_set_weak_bitmap (MonoClass *klass, int nbits, gsize *bits)
+{
+	WeakBitmapData *info = mono_class_alloc (klass, sizeof (WeakBitmapData));
+	info->nbits = nbits;
+	info->bits = bits;
```

Comment:
We should embed the bitmap at the end of WeakBitmapData and avoid the extra allocation and indirection.

Reply:
The assumption is that weak fields are not common, so the storage doesn't need to be optimized too much.